SingularityService/src/main/java/com/hubspot/singularity/resources/S3LogResource.java

```diff
@@ -98,23 +98,32 @@ public S3LogResource(RequestManager requestManager, HistoryManager historyManage
   }
 
   private Collection<String> getS3PrefixesForTask(SingularityTaskId taskId, Optional<Long> startArg, Optional<Long> endArg) {
-    SingularityTaskHistory history = getTaskHistory(taskId);
-
-    SimplifiedTaskState taskState = SingularityTaskHistoryUpdate.getCurrentState(history.getTaskUpdates());
+    Optional<SingularityTaskHistory> history = getTaskHistory(taskId);
 
     long start = taskId.getStartedAt();
     if (startArg.isPresent()) {
       start = Math.max(startArg.get(), start);
     }
 
-    long end = taskState == SimplifiedTaskState.DONE ? Iterables.getLast(history.getTaskUpdates()).getTimestamp() : System.currentTimeMillis();
+    long end = System.currentTimeMillis();
+
+    if (history.isPresent()) {
+      SimplifiedTaskState taskState = SingularityTaskHistoryUpdate.getCurrentState(history.get().getTaskUpdates());
+
+      if (taskState == SimplifiedTaskState.DONE) {
+        end = Iterables.getLast(history.get().getTaskUpdates()).getTimestamp();
+      }
+    } else if (configuration.isPresent()) {
+      end = start + configuration.get().getMissingTaskDefaultS3SearchPeriodMillis();
+    }
```

Comment:
So if the history is missing and there is no S3Configuration it searches from task start until now? Should it return an error instead?